tempodb/blocklist/poller.go

```diff
@@ -126,6 +141,14 @@ func (p *Poller) Do() (PerTenant, PerTenantCompacted, error) {
 
 		blocklist[tenantID] = newBlockList
 		compactedBlocklist[tenantID] = newCompactedBlockList
+
+		totalObjectsBlockMeta, totalObjectsCompactedBlockMeta := p.sumTotalObjects(newBlockList, newCompactedBlockList)
```

Comment:
Just a couple of questions to think about, but non blocking - 
- Should metric counts occur on the main polling function? Small but unexpected side effects (1) Longer polling time in highly multitenant envs with large blocklists, (2) will the `metricBlocklistPollDuration` drift? I guess not
- Should all components metric this? Maybe it only needs to be done by the component that creates a tenant index

I believe @joe-elliott has written most of the `Poller` code so maybe he can provide some more insight on these